### DISCUSSION TITLE
How to use proxy object?

### DISCUSSION THREAD

**Discussion by bubnenkoff** [alpinejs/alpine#2471] [1. Help] [answered]
Hello! I am not from JS world. I just need time-to-time simple web scripting. 

I tied to fetch data from web and print it in console, but it's print not data, but `proxy` object. How I can get access to data?
```

<div
    x-data="{
        posts: {},

        async getPosts() {
            this.posts = await (await fetch('http://127.0.0.1:5001/regions')).json(); // {'data': [{},{},{}] }
            console.log(this.posts);
        },

    }"
    x-init="getPosts">
</div>
```
if I want data. What should I use: ` console.log(this.posts['data']);` do not work.

![изображение](https://user-images.githubusercontent.com/6514326/145255282-d43bc379-f06c-42e5-b125-a340c11df598.png)

**Comment by danddanddand** [comment, marked as answer]
https://ryangjchandler.co.uk/posts/alpine-3-tips-and-tricks#5-unfurl-unwrap-codeproxycode-with-codealpinerawcode

**Reply by bcExpt1123** [reply]
```
<div
    x-data="{
        posts: {},

        async getPosts() {
            this.posts = await (await fetch('http://127.0.0.1:5001/regions')).json(); // {'data': [{},{},{}] }
            console.log(Alpine.raw(this.posts));
        },

    }"
    x-init="getPosts">
</div>
```

this one?

**Comment by bubnenkoff** [comment]
@danddanddand but in other situations I can simply pass `posts` to another functions?

**Reply by SimoTod** [reply]
Yes a proxy is just a transparent wrapper around the real object. Just use it as it was the object you are expecting.
